W3ChampionsStatisticService/Matches/MatchRepository.cs

```diff
@@ -148,25 +152,20 @@ private static PlayerScore CreateDetail(PlayerBlizzard playerBlizzard)
         return new PlayerScore(
             playerBlizzard.battleTag,
             playerBlizzard.unitScore,
-            playerBlizzard.heroes,
+            playerBlizzard.heroes.Select(h => new Heroes.Hero(h)).ToList(),
             playerBlizzard.heroScore,
             playerBlizzard.resourceScore,
             playerBlizzard.teamIndex);
     }
 
-    public Task<List<Matchup>> Load(
-        int season,
-        GameMode gameMode,
-        int offset = 0,
-        int pageSize = 100)
+    public Task<List<Matchup>> Load(int season, GameMode gameMode, int offset = 0, int pageSize = 100, HeroType hero = HeroType.AllFilter)
     {
         var mongoCollection = CreateCollection<Matchup>();
-        return mongoCollection
-            .Find(m => gameMode == m.GameMode && m.Season == season)
-            .SortByDescending(s => s.EndTime)
-            .Skip(offset)
-            .Limit(pageSize)
-            .ToListAsync();
+        var filter = GetLoadFilter(season, gameMode, hero);
+
+        var results = mongoCollection.Find(filter).SortByDescending(s => s.EndTime).Skip(offset).Limit(pageSize).ToListAsync();
```

Comment:
Did you double check that the index order is in the same direction as your sort?

Reply:
I don't have access to the database so I don't know what indexes exist. The public dump is missing indexes. Either way, this starts from the most recent which, I assume, is what we would want by default anyway as sorting from the oldest first makes no sense.

Reply:
The indizes are defined in the code: https://github.com/w3champions/website-backend/blob/9c4040b8daa556e7b84145a2527bf574f9e31d36/W3ChampionsStatisticService/Matches/MatchRepository.cs#L125

Reply:
But here's a screeenshot:
<img width="1473" alt="image" src="https://github.com/user-attachments/assets/8a544330-2dfb-42c6-a779-cc8868a4ade9" />

Reply:
That's only the text index for `Team1-4Players` fields. The `Season, GameMode, EndTime` index has EndTime in descending though, which this would be using as it includes Season and GameMode as well.